### DISCUSSION TITLE
이번 목표가 '상태 관리가 "왜 필요한지" 명확히 이해하자'인데 동현님은 목표 달성하셨나요?

### DISCUSSION THREAD

**Discussion by Seoin02** [React-Core-Learn/Web-Component#5] [논의]
저는 이 목표를 숙지하면서 아티클을 파헤치고 있지만, 아직 연관성을 잘 모르겠어서 논의에 올립니다.

**Comment by D5ng** [comment]
음.. 이런거 만들 때 상태관리한 것과 DOM조작한거랑 같이 비교하면서 공부했어요. 둘의 차이가 확실히 느껴지긴합니다!

지금처럼 웹 컴포넌트가 아니고 리액트를 사용하다가 바닐라로 DOM 조작해도 많이 불편하다 라는 생각이 들어요.

제가 불편하다고 느끼는 것은 직접적인 DOM조작하면 코드의 전반적인 것들을 찾아봐야했어요. 코드가 많으면 많을수록 디버깅하기가 어렵겠다는 생각이 들었어요.

반면에 상태관리로 만들면 부모에서 상태를 제어하기 때문에 제어하는 코드 부분만 보면 되기때문에 디버깅 측면에서 더 쉬웠어요

그런데 웹 컴포넌트 같은 경우는 코어 기능을 만들어야하다보니 이게 좀 어려웠네요.

**Reply by Seoin02** [reply]
저는 이번에 구현하면서 명령형이 아닌 선언적으로 프로그래밍을 하는 게 개발 속도를 개선한다는 것을 확실히 깨달았습니다. 말씀하신대로 이건 state, setState, render 등 리액트의 상태를 JS로 구현하는 것인데, 리액트 처럼 상태로 관리하는 게 아니라 바닐라 JS로 짜면 코드가 길어지고, 유지보수 하기 더 어려울 것 같아요. 자세한 내용은 회고에 남길게요!
